File: src/main/java/juuxel/lakeside/biome/BiomeTracker.java
package juuxel.lakeside.biome;

import net.minecraft.world.biome.Biome;

import java.util.Collections;
import java.util.Set;
import java.util.WeakHashMap;

public final class BiomeTracker {
    private static final Set<Biome> biomesWithLakes = Collections.newSetFromMap(new WeakHashMap<>());

    public static boolean hasLakes(Biome biome) {
        return biomesWithLakes.contains(biome);
    }

    public static void addBiomeWithLakes(Biome biome) {
        biomesWithLakes.add(biome);
    }
}

File: src/main/kotlin/juuxel/lakeside/biome/BiomeTracker.kt
package juuxel.lakeside.biome

import net.minecraft.world.biome.Biome

object BiomeTracker {
    private val biomesWithLakes = ArrayList<Biome>()

    fun addBiomeWithLakes(biome: Biome) {
        biomesWithLakes += biome
    }

    fun hasLakes(biome: Biome) = biome in biomesWithLakes
}
